src/server.c

```diff
@@ -4009,6 +4009,17 @@ int processCommand(client *c) {
         }
     }
 
+    if (!server.cluster_enabled &&
+        server.repl_replica_enable_redirect &&
+        server.masterhost &&
+        !mustObeyClient(c) &&
+        (is_write_command ||
+         (is_read_command && !(c->flags & CLIENT_READONLY)))) {
+        addReplyErrorSds(c,sdscatprintf(sdsempty(), "-REDIRECT %s:%d",
```

Comment:
Why is -1 missing here? Keep the same format as MOVED and the client can reuse the parsing code.

Reply:
-1 is not part of MOVED.

MOVED is for a slot. REDIRECT is for all writes to the node, without slot. (Only cluster has slots.)

Reply:
> MOVED is for a slot. REDIRECT is for all writes to the node, without slot. (Only cluster has slots.)

ok, that's reasonable from valkey developer.